lib/image-stream/parse-partitions.js

```diff
@@ -0,0 +1,205 @@
+/*
+ * Copyright 2017 resin.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const _ = require('lodash');
+const Bluebird = require('bluebird');
+const MBR = require('mbr');
+const GPT = require('gpt');
+
+/**
+ * @summary Maximum number of bytes to read from the stream
+ * @type {Number}
+ * @constant
+ */
+const MAX_STREAM_BYTES = 8192;
+
+/**
+ * @summary Initial number of bytes read
+ * @type {Number}
+ * @constant
+ */
+const INITIAL_LENGTH = 0;
+
+/**
+ * @summary Initial block size
+ * @type {Number}
+ * @constant
+ */
+const INITIAL_BLOCK_SIZE = 512;
+
+/**
+ * @summary Maximum block size to check for
+ * @type {Number}
+ * @constant
+ */
+const MAX_BLOCK_SIZE = 4096;
+
+/**
+ * @summary Attempt to parse the GPT from various block sizes
+ * @function
+ * @private
+ * @param {Buffer} buffer - Buffer
```

Comment:
Can we add blank lines above the first `@param` and below `@returns` for consistency purposes? (it also makes it easier to the eye to spot the signature of the function from the headers, examples, etc)